packages/api/e2e/amino.spec.ts

```diff
@@ -0,0 +1,137 @@
+import { fromBase64 } from '@cosmjs/encoding';
+import { RegenApi } from '../src/api';
+import { MsgSend } from '../src/generated/regen/ecocredit/v1/tx';
+import { makeSignDoc, StdFee } from '@cosmjs/amino/build/signdoc';
+import { Secp256k1HdWallet } from '@cosmjs/amino/build/secp256k1hdwallet';
+import { serializeSignDoc } from '@cosmjs/amino/build/signdoc';
+import { decodeSignature } from '@cosmjs/amino/build/signature';
+
+const TEST_ADDRESS = 'regen1m0qh5y4ejkz3l5n6jlrntxcqx9r0x9xjv4vpcp';
+const REDWOOD_NODE_TM_URL = 'http://redwood.regen.network:26657/';
+const TEST_FEE: StdFee = {
+  amount: [
+    {
+      denom: 'uregen',
+      amount: '5000',
+    },
+  ],
+  gas: '200000',
+};
+const TEST_MEMO = `regen-js v${process.env.npm_package_version} tests`;
+const TEST_MSG_SEND = MsgSend.fromPartial({
+  sender: TEST_ADDRESS,
+  recipient: TEST_ADDRESS,
+  credits: [
+    { batchDenom: 'C01-001-20170606-20210601-007', tradableAmount: '0.01' },
+  ],
+});
+
+const connect = async (): Promise<RegenApi> => {
+  const mnemonic =
+    'time dice choose cabbage suit panic silly cattle picture auto grab hole'; //regen1m0qh5y4ejkz3l5n6jlrntxcqx9r0x9xjv4vpcp
+
+  // Amino signer throws auth error:
+  const signer = await Secp256k1HdWallet.fromMnemonic(mnemonic, {
+    prefix: 'regen',
+  });
+
+  const accounts = await signer.getAccounts();
+
+  console.log('accounts', accounts);
+
+  // proto signer works:
+  // const signer = await DirectSecp256k1HdWallet.fromMnemonic(mnemonic, {
+  //   prefix: 'regen',
+  // });
+
+  return RegenApi.connect({
+    connection: {
+      type: 'tendermint',
+      endpoint: REDWOOD_NODE_TM_URL,
+      signer,
+    },
+  });
+};
+
+describe('RegenApi with tendermint connection', () => {
+  describe('Signing and broadcasting txs', () => {
+    it('should create a valid signature using legacy amino sign mode', async () => {
+      let txRes: DeliverTxResponse | undefined;
+      const api = await connect();
+      const msgClient = api.msgClient;
+      const mnemonic =
+        'time dice choose cabbage suit panic silly cattle picture auto grab hole'; //regen1m0qh5y4ejkz3l5n6jlrntxcqx9r0x9xjv4vpcp
+
+      // Amino signer throws auth error:
+      const signer = await Secp256k1HdWallet.fromMnemonic(mnemonic, {
+        prefix: 'regen',
+      });
+      const accounts = await signer.getAccounts();
+      const pubkey = accounts[0].pubkey;
+
+      const signDoc = makeSignDoc(
+        [{ type: MsgSend.$type, value: TEST_MSG_SEND }],
+        TEST_FEE,
+        'regen-redwood-1',
+        TEST_MEMO,
+        96,
+        2,
+      );
+
+      const { signature, signed } = await signer?.signAmino(
+        TEST_ADDRESS,
+        signDoc,
+      );
+
+      const valid = await Secp256k1.verifySignature(
+        Secp256k1Signature.fromFixedLength(fromBase64(signature.signature)),
+        sha256(serializeSignDoc(signed)),
+        pubkey,
+      );
+      expect(valid).toEqual(true);
```

Comment:
I don't think this test is super useful here. It tests the underlying functions used in cosmjs. We should only test using our own high-level `sign` and `broadcast` functions. I would propose to remove this test.